### PR TITLE
MAT-6164 Adding Measure References to generate QDM HumanReadable

### PR DESCRIPTION
**Pull request by RohitKandimalla**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-6164](https://jira.cms.gov/browse/MAT-6164)
(Optional) Related Tickets:

- Adding References to QDM HR
- Escaping free text referenceText
- referenceType is a dropdown field in UI

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/mat/cql_elm_translation/service/HumanReadableService.java [hunk 1]
```diff
@@ -130,4 +145,11 @@ private List<HumanReadablePopulationModel> buildPopulations(Group group) {
   List<HumanReadableExpressionModel> buildDefinitions(Measure measure) {
     return List.of(new HumanReadableExpressionModel());
   }
+
+  private String escapeStr(String val) {
+    if (val != null && !val.isEmpty()) {
```
Comment (adongare) [inline src/main/java/gov/cms/mat/cql_elm_translation/service/HumanReadableService.java]: maybe use `StringUtils.isNotBlank` or `StringUtils.isBlank` from apache commons

File: src/main/java/gov/cms/mat/cql_elm_translation/service/HumanReadableService.java [hunk 2]
```diff
@@ -98,6 +102,17 @@ HumanReadableMeasureInformationModel buildMeasureInfo(Measure measure) {
             measure.getMeasureMetaData() != null
                 ? measure.getMeasureMetaData().getMeasureDefinitions()
                 : null)
+        .references(
+            measure.getMeasureMetaData().getReferences().stream()
```
Comment (adongare) [inline src/main/java/gov/cms/mat/cql_elm_translation/service/HumanReadableService.java]: I just checked there might be a chance of `measure.getMeasureMetaData().getReferences()` being null instead of an empty array

Comment (RohitKandimalla) [reply src/main/java/gov/cms/mat/cql_elm_translation/service/HumanReadableService.java]: Updated it

File: src/main/java/gov/cms/mat/cql_elm_translation/service/HumanReadableService.java (hunk 1 above)
Comment (RohitKandimalla) [reply src/main/java/gov/cms/mat/cql_elm_translation/service/HumanReadableService.java]: I realized we don't even need this for referenceText, so Removed it. htmlEscape does handle nulls and empty strings. in our case, referenceText is required fields and it will never be null

**Review by adongare — APPROVED** [review]
LGTM
